src/main/java/com/feedzai/commons/sql/abstraction/batch/impl/MultithreadedBatch.java

```diff
@@ -340,6 +340,11 @@ When done, the future removes itself (if done already, all this can be skipped).
         }
     }
 
+    @Override
+    public void flushUpsert() {
+        logger.trace("Flush ignoring not available for MultithreadedBatch. Skipping ...");
+    }
```

Comment:
This cannot be that dead silent :)

If this is not implemented and if it is still being used, then something is not right -- and then, the callee logic should _fail fast_, so it can be patched as fast as possible. So, I would throw an `UnsupportedOperationException`.

(at very least, this needs to be logged and `error`)